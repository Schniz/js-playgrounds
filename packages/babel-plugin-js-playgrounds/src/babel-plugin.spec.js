const pluginTester = require("babel-plugin-tester");
const plugin = require("./babel-plugin");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  tests: {
    "Adds the playground requirement at the top": `// nothing`,
    "Wraps every expression with the playground call": `
      const x = require("x");
      const { y } = require("y");
      x(y());
      console.log(x);
    `,
    "Works with newer language features": `
      import x from 'wat';
      import {y} from 'yo';

      const f = async () => {
        let wat = await Promise.resolve("hello")
        return wat
      }
    `
  },
  formatResult: result => {
    return prettier.format(result, { parser: "babylon" });
  }
});
