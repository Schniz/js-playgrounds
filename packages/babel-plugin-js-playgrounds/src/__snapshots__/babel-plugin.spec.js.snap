// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ast-transform Adds the playground requirement at the top: Adds the playground requirement at the top 1`] = `
"
// nothing

      ↓ ↓ ↓ ↓ ↓ ↓

// nothing
var playground_fn__temp = require(\\"js-playgrounds\\");

"
`;

exports[`ast-transform Works with newer language features: Works with newer language features 1`] = `
"
import x from 'wat';
import {y} from 'yo';

const f = async () => {
  let wat = await Promise.resolve(\\"hello\\")
  return wat
}

      ↓ ↓ ↓ ↓ ↓ ↓

var playground_fn__temp = require(\\"js-playgrounds\\");
import x from \\"wat\\";
import { y } from \\"yo\\";

const f = playground_fn__temp(
  {
    start: {
      line: \\"4\\",
      column: \\"10\\"
    },
    end: {
      line: \\"7\\",
      column: \\"1\\"
    }
  },
  async () => {
    let wat = playground_fn__temp(
      {
        start: {
          line: \\"5\\",
          column: \\"12\\"
        },
        end: {
          line: \\"5\\",
          column: \\"42\\"
        }
      },
      await Promise.resolve(\\"hello\\")
    );
    return wat;
  }
);

"
`;

exports[`ast-transform Wraps every expression with the playground call: Wraps every expression with the playground call 1`] = `
"
const x = require(\\"x\\");
const { y } = require(\\"y\\");
x(y());
console.log(x);

      ↓ ↓ ↓ ↓ ↓ ↓

var playground_fn__temp = require(\\"js-playgrounds\\");
const x = playground_fn__temp(
  {
    start: {
      line: \\"1\\",
      column: \\"10\\"
    },
    end: {
      line: \\"1\\",
      column: \\"22\\"
    }
  },
  require(\\"x\\")
);
const { y } = playground_fn__temp(
  {
    start: {
      line: \\"2\\",
      column: \\"14\\"
    },
    end: {
      line: \\"2\\",
      column: \\"26\\"
    }
  },
  require(\\"y\\")
);
playground_fn__temp(
  {
    start: {
      line: \\"3\\",
      column: \\"0\\"
    },
    end: {
      line: \\"3\\",
      column: \\"6\\"
    }
  },
  x(y())
);
playground_fn__temp(
  {
    start: {
      line: \\"4\\",
      column: \\"0\\"
    },
    end: {
      line: \\"4\\",
      column: \\"14\\"
    }
  },
  console.log(x)
);

"
`;
